### PR TITLE
rename DataLevel tableId() to metaTableId(), doc updates

### PR DESCRIPTION
**Pull request by EdColeman**
Renames the DataLevel method tableId to metaTableId to help clarify that the id returned is the reference to the meta data that contain information for that level (USER points to metadata table, METADATA points to ROOT and ROOT point to ZooKeeper)

Also adds some javadoc to emphasize the level, meta data relationship for forLevel and forTable in TabletsMetadata

### CODE REVIEW

File: core/src/main/java/org/apache/accumulo/core/metadata/schema/TabletsMetadata.java [hunk 1]
```diff
@@ -319,13 +319,24 @@ public Options fetch(ColumnType... colsToFetch) {
...
     public Options forLevel(DataLevel level) {
       this.level = level;
       this.range = TabletsSection.getRange();
       return this;
     }
 
+    /**
+     * Set the range to the data level that contains the metadata references for the provided
+     * TableId. For example, if the TableId is the metadata table id, then the table range will be
+     * root metadata that points to the metadata. Likewise, if the TableId is a user table, then the
```
Comment (ddanielr) [inline core/src/main/java/org/apache/accumulo/core/metadata/schema/TabletsMetadata.java:337]: This comment is a bit hard to follow. 

Can the metadata table location references be simplified
like `root metadata range` vs `root metadata that points to the metadata`

Comment (EdColeman) [reply core/src/main/java/org/apache/accumulo/core/metadata/schema/TabletsMetadata.java:337]: I'm open to suggestions.  I was trying to convey that user -> metadata, metadata -> root, and root -> ZooKeeper.  One issue is that metadata is overloaded.  When we say metadata, is it the table? The information for a level that references point to the next level?  Even that last sentence is awkward for me, so if someone has any suggestions I'll happily hit commit suggestion and move on.

File: core/src/main/java/org/apache/accumulo/core/metadata/schema/TabletsMetadata.java [hunk 2]
```diff
@@ -319,13 +319,24 @@ public Options fetch(ColumnType... colsToFetch) {
       return this;
     }
 
+    /**
+     * Set the range to the table range that stores the provided level's metadata. For example
+     * {@link DataLevel#USER} will set the range to the metadata table. {@link DataLevel#METADATA}
+     * will set the range to the root table.
+     */
```
Comment (keith-turner) [inline core/src/main/java/org/apache/accumulo/core/metadata/schema/TabletsMetadata.java:326]:
```suggestion
    /**
     * For a given data level, read all of its tablets metadata.  For DataLevel.USER this will read tablet metadata from the accumulo.metadata table.  For DataLevel.METADATA this will read tablet metadata from the accumulo.root table.  For DataLevel.ROOT this will read tablet metadata from Zookeeper.
     */
```

File: core/src/main/java/org/apache/accumulo/core/metadata/schema/TabletsMetadata.java [hunk 3]
```diff
@@ -319,13 +319,24 @@ public Options fetch(ColumnType... colsToFetch) {
...
       this.range = TabletsSection.getRange();
       return this;
     }
 
+    /**
+     * Set the range to the data level that contains the metadata references for the provided
+     * TableId. For example, if the TableId is the metadata table id, then the table range will be
+     * root metadata that points to the metadata. Likewise, if the TableId is a user table, then the
+     * table range will be the metadata table range that points to the user table.
+     */
```
Comment (keith-turner) [inline core/src/main/java/org/apache/accumulo/core/metadata/schema/TabletsMetadata.java:339]:
```suggestion
    /**
     * For a given table read all of its tablet metadata.  If the table id  is for a user table, then its metadata will be read from its section in the accumulo.metadata table. If the table is is for the accumulo.metadata table, then its metadata will be read from the accumulo.root table.  If the table id is for the accumulo.root table, then its metadata will be read from zookeeper.
     */
```

**Review by keith-turner — COMMENTED** [review]
Made some suggested javadoc updates.  Since there are only three levels, we can just list exactly what is done for all three.  Thinking this makes the docs more straightforward.

**Comment by ddanielr** [comment]
@EdColeman once Keith's documentation suggestions are committed, I think this is good to merge.

**Review by ddanielr — APPROVED** [review]

**Review by keith-turner — APPROVED** [review]
